pkgs/development/python-modules/nifty8/default.nix

```
@@ -0,0 +1,86 @@
+{
+  config,
+  lib,
+  stdenv,
+  buildPythonPackage,
+  fetchFromGitLab,
+
+  cudaSupport ? config.cudaSupport,
+
+  # build-system
+  setuptools,
+
+  # dependencies
+  astropy,
+  ducc0,
+  h5py,
+  jax,
+  jaxlib,
+  matplotlib,
+  mpi,
+  mpi4py,
+  numpy,
+  scipy,
+
+  # test
+  pytestCheckHook,
+  mpiCheckPhaseHook,
+  openssh,
+}:
+
+buildPythonPackage rec {
+  pname = "nifty8";
+  version = "8.5.2";
+  pyproject = true;
+
+  src = fetchFromGitLab {
+    domain = "gitlab.mpcdf.mpg.de";
+    owner = "ift";
+    repo = "nifty";
+    rev = "v${version}";
+    hash = "sha256-EWsJX+iqKOhQXEWlQfYUiPYqyfOfrwLtbI+DVn7vCQI=";
+  };
+
+  build-system = [ setuptools ];
+
+  dependencies = [
+    astropy
+    ducc0
+    h5py
+    jax
+    (jaxlib.override { inherit cudaSupport; })
+    matplotlib
+    mpi4py
+    mpi
+    numpy
+    scipy
+  ];
+
+  nativeCheckInputs = [
+    pytestCheckHook
+    mpiCheckPhaseHook
+    openssh
+  ];
+
+  checkPhase = ''
+    runHook preCheck
+
+    python3 -m pytest test
+
+    if [ "${stdenv.hostPlatform.system}" != "aarch64-linux" ] && \
+       [ "${stdenv.hostPlatform.system}" != "x86_64-darwin" ]; then
```

```suggestion
    if [ "${stdenv.buildPlatform.system}" != "aarch64-linux" ] && \
       [ "${stdenv.buildPlatform.system}" != "x86_64-darwin" ]; then
```
Though it doesn't really matter since tests are only run for build == host anyway.